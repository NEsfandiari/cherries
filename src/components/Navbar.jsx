import React, { Component } from 'react'
import Styled from 'styled-components'
import PropTypes from 'prop-types'
import { NavButtons, BannerPromo } from './molecules'
import CartSidebar from './CartSidebar.jsx'
import MobileSidebar from './MobileSidebar.jsx'
import { CartConsumer } from '../containers/CartContext'
import 'futura-font/styles.css'

const NavContainer = Styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  z-index: 2;
`

class Navbar extends Component {
  //TODO: Mobile work
  // state = {
  //   mobileSidebar: false,
  //   mobileDisplayFix: false,
  // }
  static contextTypes = {
    firebase: PropTypes.object,
  }
  render() {
    const {
      curUser,
      sidebar,
      displayFix,
      handleSidebar,
      handleBannerMargin,
    } = this.props
    //TODO: Mobile work
    // const { mobileSidebar, mobileDisplayFix } = this.state
    return (
      <div>
        {/* //TODO: Mobile work
        <MobileSidebar
          display={mobileSidebar}
          handleMobileSidebar={this.handleMobileSidebar}
          logOut={this.logOut}
          curUser={curUser}
          handleSidebar={handleSidebar}
          mobileDisplayFix={mobileDisplayFix}
        /> */}
        <NavContainer>
          <BannerPromo handleBannerMargin={handleBannerMargin} />
          <NavButtons
            curUser={curUser}
            handleMobileSidebar={this.handleMobileSidebar}
            handleSidebar={handleSidebar}
            logOut={this.logOut}
          />
        </NavContainer>
        <CartConsumer>
          {cartContext => (
            <CartSidebar
              cart={cartContext.cart}
              handleCart={cartContext.handleCart}
              curUser={curUser}
              display={sidebar}
              handleSidebar={handleSidebar}
              displayFix={displayFix}
            />
          )}
        </CartConsumer>
      </div>
    )
  }
}
