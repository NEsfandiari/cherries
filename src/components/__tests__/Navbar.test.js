import React from 'react'
import renderer from 'react-test-renderer'
import { Navbar, PureNavbar } from '../Navbar'

describe('Navbar', () => {
  it('renders correctly', () => {
    const data = {
      contentfulHomePage: {
        navHelpIcon: {
          fluid: {
            src:
              '//images.ctfassets.net/4yejk6rho2rp/6u7gsl9D3Oo46iYwW04y8c/66b1a6434576a65e23d5501ee03fb1a8/help-icon.png?w=800&q=50',
          },
        },
        navUserIcon: {
          fluid: {
            src:
              '//images.ctfassets.net/4yejk6rho2rp/8WGaI6nf8suy8YUQYsois/17e7023c4ad873b82a22bf6704de790e/user-icon.png?w=800&q=50',
          },
        },
        navCartIcon: {
          fluid: {
            src:
              '//images.ctfassets.net/4yejk6rho2rp/6NiPYpu9FYKqs4CW6QAKCi/cd5b1b98fe86f6fe2f264acbdf39dbdb/shopping-bag-icon.png?w=800&q=50',
          },
        },
        navSearchIcon: {
          fluid: {
            src:
              '//images.ctfassets.net/4yejk6rho2rp/1Kjc0inKTe0AEomGUy2iu/bc85cf1f149b25cb2513069c4a64846d/magnifying-glass__2_.png?w=800&q=50',
          },
        },
        navCherriesIcon: {
          fluid: {
            src:
              '//images.ctfassets.net/4yejk6rho2rp/vXHBTTvYRwKKsukOMKEQi/b1c8e9091314f28f5b66f0770f1da457/cherries-logo.png?w=800&q=50',
          },
        },
        bannerPromoText:
          'This is the body text of a banner announcement. Could be about prods etc etc etc.',
        bannerPromoTitle: 'Announcements! ',
        userLinks: {
          data: [
            {
              dropdownLinks: [
                {
                  name: 'Sign Up',
                  route: 'signup',
                },
                {
                  name: 'Log In',
                  route: 'login',
                },
              ],
            },
            {
              dropdownLinks: [
                {
                  name: 'Account',
                  route: 'account',
                },
                {
                  name: 'Log Out',
                  route: '',
                },
              ],
            },
          ],
        },
        helpLinks: {
          data: [
            {
              dropdownLinks: [
                {
                  name: 'FAQ',
                  route: 'faq',
                },
                {
                  name: 'Help',
                  route: 'faq',
                },
                {
                  name: 'Returns',
                  route: 'faq',
                },
                {
                  name: 'team@lipslut.com',
                  route: 'contact',
                },
              ],
            },
          ],
        },
      },
    }
    const tree = renderer.create(<PureNavbar data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
