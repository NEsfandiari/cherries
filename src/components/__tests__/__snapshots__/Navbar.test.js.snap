// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div
  className="sc-hSdWYo kVNLLi"
>
  <div
    className="sc-bwzfXH bSaftk"
  >
    <div
      className="banner-text sc-htpNat XNTOc"
    >
      Announcements! 
    </div>
    <div
      className="banner-text sc-bxivhb hFWgEG"
    >
      This is the body text of a banner announcement. Could be about prods etc etc etc.
    </div>
    <div
      className="sc-ifAKCX gbLoNY"
    >
      <svg
        fill="currentColor"
        height={30}
        onClick={[Function]}
        size={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width={30}
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </div>
     
  </div>
  <div
    className="sc-dxgOiQ dpHeJh"
  >
    <div
      className="leftNav"
    >
      <img
        alt="search-icon"
        onClick={[Function]}
        src="//images.ctfassets.net/4yejk6rho2rp/1Kjc0inKTe0AEomGUy2iu/bc85cf1f149b25cb2513069c4a64846d/magnifying-glass__2_.png?w=800&q=50"
        style={
          Object {
            "margin": 10,
            "maxHeight": 28,
            "maxWidth": 28,
          }
        }
      />
    </div>
    <div
      className="logo"
    >
      <a
        href="/"
      >
        <img
          alt="Cherries Logo"
          src="//images.ctfassets.net/4yejk6rho2rp/vXHBTTvYRwKKsukOMKEQi/b1c8e9091314f28f5b66f0770f1da457/cherries-logo.png?w=800&q=50"
          style={
            Object {
              "margin": "0 auto",
              "maxHeight": 48,
              "maxWidth": 48,
            }
          }
        />
      </a>
    </div>
    <div
      className="rightNav"
    >
      <div
        className="sc-EHOje guXxGN"
        onMouseLeave={[Function]}
      >
        <a
          className="dropdown sc-htoDjs hfJzAp"
          href=""
          onMouseEnter={[Function]}
        >
          <img
            alt="nav-bar-icon"
            src="//images.ctfassets.net/4yejk6rho2rp/6u7gsl9D3Oo46iYwW04y8c/66b1a6434576a65e23d5501ee03fb1a8/help-icon.png?w=800&q=50"
            style={
              Object {
                "margin": 5,
                "maxHeight": 28,
                "maxWidth": 28,
              }
            }
          />
        </a>
        <div
          className="dropdown-content"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="links"
          >
            <a
              className="sc-htoDjs hfJzAp"
              href="faq"
            >
              FAQ
            </a>
            <a
              className="sc-htoDjs hfJzAp"
              href="faq"
            >
              Help
            </a>
            <a
              className="sc-htoDjs hfJzAp"
              href="faq"
            >
              Returns
            </a>
            <a
              className="sc-htoDjs hfJzAp"
              href="contact"
            >
              team@lipslut.com
            </a>
          </div>
        </div>
      </div>
      <div
        className="sc-EHOje guXxGN"
        onMouseLeave={[Function]}
      >
        <a
          className="dropdown sc-htoDjs hfJzAp"
          href=""
          onMouseEnter={[Function]}
        >
          <img
            alt="nav-bar-icon"
            src="//images.ctfassets.net/4yejk6rho2rp/8WGaI6nf8suy8YUQYsois/17e7023c4ad873b82a22bf6704de790e/user-icon.png?w=800&q=50"
            style={
              Object {
                "margin": 5,
                "maxHeight": 28,
                "maxWidth": 28,
              }
            }
          />
        </a>
        <div
          className="dropdown-content"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="links"
          >
            <a
              className="sc-htoDjs hfJzAp"
              href="signup"
            >
              Sign Up
            </a>
            <a
              className="sc-htoDjs hfJzAp"
              href="login"
            >
              Log In
            </a>
          </div>
        </div>
      </div>
      <div
        className="sc-gZMcBi fMZEJd"
      >
        <img
          alt="search-icon"
          src="//images.ctfassets.net/4yejk6rho2rp/6NiPYpu9FYKqs4CW6QAKCi/cd5b1b98fe86f6fe2f264acbdf39dbdb/shopping-bag-icon.png?w=800&q=50"
          style={
            Object {
              "marginBottom": 5,
              "marginLeft": 10,
              "marginRight": 20,
              "maxHeight": 26,
              "maxWidth": 23.92,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
